pandas/core/construction.py

```diff
@@ -305,38 +306,38 @@ def array(
         elif inferred_dtype.startswith("datetime"):
             # datetime, datetime64
             try:
-                return DatetimeArray._from_sequence(data, copy=copy)
+                return DatetimeArray._from_sequence(_data, copy=copy)
             except ValueError:
                 # Mixture of timezones, fall back to PandasArray
                 pass
 
         elif inferred_dtype.startswith("timedelta"):
             # timedelta, timedelta64
-            return TimedeltaArray._from_sequence(data, copy=copy)
+            return TimedeltaArray._from_sequence(_data, copy=copy)
 
         elif inferred_dtype == "string":
-            return StringArray._from_sequence(data, copy=copy)
+            return StringArray._from_sequence(_data, copy=copy)
 
         elif inferred_dtype == "integer":
-            return IntegerArray._from_sequence(data, copy=copy)
+            return IntegerArray._from_sequence(_data, copy=copy)
 
         elif inferred_dtype == "boolean":
-            return BooleanArray._from_sequence(data, copy=copy)
+            return BooleanArray._from_sequence(_data, copy=copy)
 
     # Pandas overrides NumPy for
     #   1. datetime64[ns]
     #   2. timedelta64[ns]
     # so that a DatetimeArray is returned.
     if is_datetime64_ns_dtype(dtype):
-        return DatetimeArray._from_sequence(data, dtype=dtype, copy=copy)
+        return DatetimeArray._from_sequence(_data, dtype=dtype, copy=copy)
     elif is_timedelta64_ns_dtype(dtype):
-        return TimedeltaArray._from_sequence(data, dtype=dtype, copy=copy)
+        return TimedeltaArray._from_sequence(_data, dtype=dtype, copy=copy)
 
-    result = PandasArray._from_sequence(data, dtype=dtype, copy=copy)
+    result = PandasArray._from_sequence(_data, dtype=dtype, copy=copy)
     return result
 
 
-def extract_array(obj, extract_numpy: bool = False):
+def extract_array(obj: Any, extract_numpy: bool = False) -> Union[Any, ArrayLike]:
```

Comment:
does it work to make the Any a `Sequence[object]` ?

Reply:
extract_array accepts any object and returns the object unchanged if can't extract array.

https://github.com/pandas-dev/pandas/blob/8dd3d879d60892525c1c7f02ee3659b00f24fa60/pandas/core/construction.py#L381-L387

we could look into changing this (raise instead of returning object?) but would not change the numpy array resolving to Any issues.

Reply:
thinking some more, will revert this for now.